lib/segment/src/common/utils.rs

```diff
@@ -316,6 +317,109 @@ pub fn remove_value_from_json_map(
     }
 }
 
+/// Insert value at a given JSON path  in JSON array
+pub fn insert_at_array_path(
+    array_path: &str,
+    array_index: Option<u32>,
+    rest_path: Option<&str>,
```

Comment:
nit: this variable name is kinda confusing
```suggestion
    rest_of_path: Option<&str>,
```
My dumb ass was thinking it was something related to REST path, until I saw the context of `insert_at_path`

Reply:
Was confused by this as well at first. Good suggestion.

Reply:
Thanks for the suggestion, I now realize that it can be weird :+1: